### PR TITLE
Release/v3.3.6

### PR DESCRIPTION
**Pull request by fzipi**
## Version 3.3.6 - 2024-08-27

Important changes:

* Backport fix for 3MU-240701-1 - catch invalid character in multipart headers via new rule 922130 (Ervin Hegedus, Felipe Zipitría)
* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 (Felipe Zipitria)

### CODE REVIEW

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 1]
```diff
@@ -623,7 +623,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     tag:'PCI/6.5.2',\
     tag:'paranoia-level/2',\
     ctl:auditLogParts=+E,\
-    ver:'OWASP_CRS/3.3.5',\
+    ver:'OWASP_CRS/3.3.6',\
```
Comment (airween) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:626]: This rule above in line 614 uses both `(?i)` in regexp and `t:lowercase` - do we want to care this? (In CRS4 we do not allow this)

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 2]
```diff
@@ -83,20 +83,19 @@ SecRule REQUEST_LINE "!@rx ^(?i:(?:[a-z]{3,10}\s+(?:\w{3,7}?://[\w\-\./]*(?::\d+
...
 #
 # Not supported by re2 (?<!re).
 #
 # -=[ References ]=-
-# https://www.owasp.org/index.php/ModSecurity_CRS_RuleID-960000
 # http://www.ietf.org/rfc/rfc2183.txt
 #
-SecRule FILES_NAMES|FILES "@rx (?<!&(?:[aAoOuUyY]uml)|&(?:[aAeEiIoOuU]circ)|&(?:[eEiIoOuUyY]acute)|&(?:[aAeEiIoOuU]grave)|&(?:[cC]cedil)|&(?:[aAnNoO]tilde)|&(?:amp)|&(?:apos));|['\"=]" \
+SecRule FILES_NAMES|FILES "@rx (?<!&(?:[aAoOuUyY]uml)|&(?:[aAeEiIoOuU]circ)|&(?:[eEiIoOuUyY]acute)|&(?:[aAeEiIoOuU]grave)|&(?:[cC]cedil)|&(?:[aAnNoO]tilde)|&(?:amp)|&(?:apos));|['\"=\x5c]" \
     "id:920120,\
```
Comment (RedXanadu) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf:99]: I think we need to make users aware that this rule has changed. Maybe a line in the CHANGES file to note this?

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 3]
```diff
@@ -1362,15 +1361,15 @@ SecRule &REQUEST_HEADERS:User-Agent "@eq 0" \
...
     severity:'NOTICE',\
     setvar:'tx.anomaly_score_pl2=+%{tx.notice_anomaly_score}'"
 
 
 #
 # PL2: This is a stricter sibling of 920120.
 #
-SecRule FILES_NAMES|FILES "@rx ['\";=]" \
+SecRule FILES_NAMES|FILES "@rx ['\";=\x5c]" \
     "id:920121,\
```
Comment (RedXanadu) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf:1373]: The same for this rule: "updated patterns for rules X, Y, and Z to combat multipart bypass" or something?

**Review by RedXanadu — CHANGES_REQUESTED** [review]
I think that, because v3.3.x is an "LTS" / stable release line, we must explicitly tell users that we've changed regex patterns in rules X and Y in the CHANGES log.

Other than that, this looks good to me. Great work!

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf (hunk 1 above)
Comment (fzipi) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:626]: I don't think we should change that.

File: CHANGES.md [hunk 4]
```diff
@@ -6,6 +6,14 @@
   or the CRS Google Group at
 * <https://groups.google.com/a/owasp.org/forum/#!forum/modsecurity-core-rule-set-project>
 
+
+## Version 3.3.6 - 2024-08-27
+
+Important changes:
+
+* Backport fix for 3MU-240701-1 - catch invalid character in multipart headers via new rule 922130 (Ervin Hegedus, Felipe Zipitría)
+* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 (Felipe Zipitria)
```
Comment (fzipi) [inline CHANGES.md:15]:
```suggestion
* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 - updated rule 920120 (Felipe Zipitria)
```

File: CHANGES.md [hunk 5]
```diff
@@ -6,6 +6,14 @@
   or the CRS Google Group at
 * <https://groups.google.com/a/owasp.org/forum/#!forum/modsecurity-core-rule-set-project>
 
+
+## Version 3.3.6 - 2024-08-27
+
+Important changes:
+
+* Backport fix for 3MU-240701-1 - catch invalid character in multipart headers via new rule 922130 (Ervin Hegedus, Felipe Zipitría)
+* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 - updated rule 920120 (Felipe Zipitria)
```
Comment (fzipi) [inline CHANGES.md:15]:
```suggestion
* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 - updated rule 920120 - pl1, 920121 - pl2 (Felipe Zipitria)
```

File: CHANGES.md [hunk 6]
```diff
@@ -6,6 +6,14 @@
   or the CRS Google Group at
 * <https://groups.google.com/a/owasp.org/forum/#!forum/modsecurity-core-rule-set-project>
 
+
+## Version 3.3.6 - 2024-08-27
+
+Important changes:
+
+* Backport fix for 3MU-240701-1 - catch invalid character in multipart headers via new rule 922130 (Ervin Hegedus, Felipe Zipitría)
+* Backport fix for 3MU-240701-2 - prevent using backslash in file names from v4 - updated rule 920120 - pl1, 920121 - pl2 (Felipe Zipitria)
```
Comment (fzipi) [inline CHANGES.md:15]: Would this text be enough @RedXanadu?

**Review by airween — APPROVED** [review]

File: CHANGES.md (hunk 6 above)
Comment (RedXanadu) [reply CHANGES.md:15]: Perfect.

**Review by RedXanadu — APPROVED** [review]

**Pull request merged**